src/Ztm.WebApi/AddressPools/ReceivingAddressPool.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Ztm.WebApi.AddressPools
+{
+    public sealed class ReceivingAddressPool : IReceivingAddressPool
+    {
+        readonly IAddressGenerator generator;
+        readonly IReceivingAddressStorage storage;
+        readonly IAddressChoser choser;
+
+        readonly ReaderWriterLockSlim storageLock;
```

Comment:
Firstly, I decide to move logic to pool because need to simplify storage. However, we need to support case that addresses in storage is shares between other user. I decide to move the logic back to storage by reverting a commit.